front/lib/models/assistant/configuration.ts

```diff
@@ -0,0 +1,350 @@
+import {
+  CreationOptional,
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { front_sequelize } from "@app/lib/databases";
+import { DataSource } from "@app/lib/models/data_source";
+import { Workspace } from "@app/lib/models/workspace";
+import { TimeframeUnit } from "@app/types/assistant/actions/retrieval";
+import {
+  AgentConfigurationScope,
+  AgentConfigurationStatus,
+} from "@app/types/assistant/configuration";
+
+/**
+ * Agent configuration
+ */
+export class AgentConfiguration extends Model<
+  InferAttributes<AgentConfiguration>,
+  InferCreationAttributes<AgentConfiguration>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare sId: string;
+  declare status: AgentConfigurationStatus;
+  declare name: string;
+  declare pictureUrl: string | null;
+  declare scope: AgentConfigurationScope;
+
+  declare workspaceId: ForeignKey<Workspace["id"]> | null; // null = it's a global agent
+  declare generationId: ForeignKey<AgentGenerationConfiguration["id"]> | null;
```

Comment:
generationConfigurationId
retrievalConfigurationId

It's a bit long but less chance to get confused by thinking it's the "result" ?